rest/handler.go

```diff
@@ -35,62 +38,86 @@ func CreateHandler(ctx context.Context, prometheusURL string) (http.Handler, err
 		return nil, err
 	}
 
-	return &langserverHandler{langserver: langserver}, nil
-}
+	ls := &langserverHandler{langserver: langserver}
+	ls.m = make(map[string]http.Handler)
+	ls.createDefaultHandlers()
 
-type langserverHandler struct {
-	langserver     langserver.HeadlessServer
-	requestCounter int64
+	return ls, nil
 }
 
-func (h *langserverHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	var subHandler func(w http.ResponseWriter, r *http.Request, s langserver.HeadlessServer, requestID string)
-
-	requestID := fmt.Sprint(atomic.AddInt64(&h.requestCounter, 1), ".promql")
-
-	switch r.URL.Path {
-	case "/diagnostics":
-		subHandler = diagnosticsHandler
-	case "/hover":
-		subHandler = hoverHandler
-	case "/completion":
-		subHandler = completionHandler
-	case "/signatureHelp":
-		subHandler = signatureHelpHandler
-	default:
-		http.NotFound(w, r)
-		return
+// CreateInstHandler creates an instrumented http.Handler for the PromQL langserver REST API.
+//
+// Expects the URL of a Prometheus server as the second argument and a Registry as third argument.
+func CreateInstHandler(ctx context.Context, prometheusURL string, r *prometheus.Registry) (http.Handler, error) {
+	langserver, err := langserver.CreateHeadlessServer(ctx, prometheusURL)
+	if err != nil {
+		return nil, err
 	}
 
-	exprs, ok := r.URL.Query()["expr"]
+	ls := &langserverHandler{langserver: langserver}
+	ls.m = make(map[string]http.Handler)
+	ls.createDefaultInstHandlers(r)
 
-	if !ok || len(exprs) == 0 {
-		http.Error(w, "Param expr is not specified", 400)
-		return
+	return ls, nil
+}
+
+func (h *langserverHandler) createDefaultHandlers() {
```

Comment:
`createDefaultHandlers` and `createDefaultInstHandlers` could be one function with some 

```
if r != nil {
...
```

Also, why does the function name contain the word "Default"?

Reply:
there is a Handle function that set handlers just incase functionality grows in the future. the functions created by the functions are compulsory handler needed for the endpoint exposed hence the name default . should i exclude the default then?

Reply:
> should i exclude the default then?
Yes.